featurebyte/models/offline_store_ingest_query.py

```diff
@@ -273,8 +277,91 @@ def generate_on_demand_feature_view_code(
             code_generator.add_statements(statements=[statements])
 
         codes = code_generator.generate(
+            to_format=True,
             input_df_name=input_df_name,
             output_df_name=output_df_name,
             function_name=function_name,
         )
         return codes
+
+    def _generate_on_demand_feature_function_code_state(
+        self,
+        output_dtype: DBVarType,
+        sql_function_name: str = "odf_func",
+        sql_input_var_prefix: str = "x",
+        sql_request_input_var_prefix: str = "r",
+        sql_comment: str = "",
+        function_name: str = "on_demand_feature_function",
+        input_var_prefix: str = "col",
+        request_input_var_prefix: str = "request_col",
+    ) -> OnDemandFeatureFunctionGlobalState:
+        if not self.is_decomposed:
+            raise ValueError(
+                "Cannot generate on demand feature function code for non-decomposed query graph"
+            )
+
+        node = self.graph.get_node_by_name(self.node_name)
+        codegen_state = OnDemandFeatureFunctionExtractor(graph=self.graph).extract(
+            node=node,
+            sql_function_name=sql_function_name,
+            sql_input_var_prefix=sql_input_var_prefix,
+            sql_request_input_var_prefix=sql_request_input_var_prefix,
+            sql_comment=sql_comment,
+            function_name=function_name,
+            input_var_prefix=input_var_prefix,
+            request_input_var_prefix=request_input_var_prefix,
+            output_dtype=output_dtype,
+        )
+        return codegen_state
+
+    def generate_on_demand_feature_function_code(
+        self,
+        output_dtype: DBVarType,
+        to_sql: bool = False,
+        sql_function_name: str = "odf_func",
+        sql_input_var_prefix: str = "x",
+        sql_request_input_var_prefix: str = "r",
+        sql_comment: str = "",
```

Comment:
Do these parameters apply when `to_sql` is False?

Reply:
Nope. Is used only when it is `True`.